torchvision/prototype/transforms/_geometry.py

```diff
@@ -448,43 +449,62 @@ def __init__(
     def _get_params(self, sample: Any) -> Dict[str, Any]:
         image = query_image(sample)
         _, height, width = get_image_dimensions(image)
+
+        if self.padding is not None:
+            # update height, width with static padding data
+            padding = self.padding
+            if isinstance(padding, Sequence):
+                padding = list(padding)
+            pad_left, pad_right, pad_top, pad_bottom = _parse_pad_padding(padding)
+            height += pad_top + pad_bottom
+            width += pad_left + pad_right
+
         output_height, output_width = self.size
+        # We have to store maybe padded image size for pad_if_needed branch in _transform
+        input_height, input_width = height, width
+
+        if self.pad_if_needed:
+            # pad width if needed
+            if width < output_width:
+                width += 2 * (output_width - width)
+            # pad height if needed
+            if height < output_height:
+                height += 2 * (output_height - height)
```

Comment:
@datumbox to explain why I didn't implemented that earlier. What I do not like in this approach is that we have to precompute image size before doing any manipulations on the data...

Reply:
Yeap understood. At the same time, this re-computation issue was discussed previously with the rotate and we decided it was OK to do it. So I'm also OK doing it here. We can review if this turns out to be a slowdown.

Reply:
> At the same time, this re-computation issue was discussed previously with the rotate and we decided it was OK to do it

If you talk about this:
https://github.com/pytorch/vision/blob/c31db6d640fc92fcabbd8ae7212eb979096ca0be/torchvision/prototype/features/_bounding_box.py#L174-L182
IMO it is a bit another degree of recomputed things

Reply:
> IMO it is a bit another degree of recomputed things

Just to make sure I get your point, what you mean another degree? Which one in your opinion is worse/more-costly?

Reply:
In case of rotate, we recompute due to low-level ops that do not provide meta-data.
In case of random crop, we show that if combined transformation needs dynamic params (next op's params depend on previous op's result), we have to precompute the whole pipeline in `_get_params` before executing actual ops on real data.

Reply:
OK so the reason for the recomputation is different but the practice remains the same. Does one of the two practices concerns you more? I just want to understand if you have concerns over the use of this idiom.

Reply:
No blocking concerns right now. Let's move on :)